### PR TITLE
refactor: refactor sharing functionality into shared mixin

### PR DESCRIPTION
**Pull request by eddieferrer**
There are currently 3 components which share to social media

- SocialShareV2 (ML thanks, thank-page-v2 layout on the thanks page)
- KvSocialShareButton (Borrower profile)
- ThanksPageShare (mars share layout on the thanks page)

This combines some of the shared functionality of these components into one place.

### CODE REVIEW

File: src/components/Thanks/ThanksPageShare.vue [hunk 1]
```diff
@@ -269,8 +269,6 @@ export default {
 		shareLink() {
 			const base = `https://${this.$appConfig.host}`;
 			const args = {
-				utm_campaign: this.utmCampaign,
```
Comment (eddieferrer) [inline src/components/Thanks/ThanksPageShare.vue:272]: These will be added in the mixin and should not be added here

File: src/plugins/social-sharing-mixin.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+			}
+			window.open(
+				destinationUrl,
+				'intent',
+				// eslint-disable-next-line max-len
+				`scrollbars=yes,resizable=yes,toolbar=no,location=yes,width=${width},height=${height},left=${left},top=${top}`
+			);
+			this.$showTipMsg(thanksText);
+		},
+		twitterShareUrl({ utmCampaign, utmContent }) {
```
Comment (eddieferrer) [inline src/plugins/social-sharing-mixin.js:49]: The 3 functions work the same way. Take the url to be shared, first, add utm params. Then run that url as the input param for adding the params required by the social media platform.

File: src/plugins/social-sharing-mixin.js [hunk 3]
```diff
@@ -0,0 +1,136 @@
+import { getFullUrl } from '@/util/urlUtils';
+import clipboardCopy from 'clipboard-copy';
+
+/* This mixin depends on the component using it having the
+* following data or computed variables available:
+* this.shareMessage and this.shareLink
```
Comment (eddieferrer) [inline src/plugins/social-sharing-mixin.js:6]: I tried to make this as generic as possible for this first pass, but it still depends on shareMessage and shareLink from the component

File: src/util/urlUtils.js [hunk 4]
```diff
@@ -48,10 +48,17 @@ export function isExcludedUrl(urlArray, route) {
  */
 export function getFullUrl(base, args) {
 	if (!args || Object.keys(args).length === 0) return base;
+
+	// remove hash portion of url if present
```
Comment (eddieferrer) [inline src/util/urlUtils.js:52]: Added functionality and additional tests. Function now supports args that are empty/null/undefined. These are skipped in the generation of the query params. Function also now supports hashes. For example `kiva.org#updates` should have query params inserted before the hashes, and produce something like `kiva.org?query1=true#updates`

**Review by dyersituations — APPROVED** [review]
The share link methods didn't have unit tests before, but those might be good to add for the new mixin at some point. Some generally complicated logic in the mixin.

**Pull request merged**